### PR TITLE
feat(i18n): bump i18next and add typings

### PR DESCRIPTION
**Pull request by ckniffen**
## High Level Overview of Change

Newer versions of i18next necessitated improving the typings of translations. As a result the build now fail if a translation is referenced that is not in the `en-US` file.  This already caught several missing and/or misspelled keys.

The `context` option of `t` is now being leveraged for enum based translations.  Other dynamic places had to use `as any` to get around typing errors.  Luckily there are only a few and in the future they could also leverage context.

Replaces #690

### Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Refactor (non-breaking change that only restructures code)
- [x] Tests (You added tests for code that already exists, or your new feature included in this PR)

### CODE REVIEW

File: public/locales/en-US/translations.json [hunk 1]
```diff
@@ -141,47 +146,47 @@
...
-  "transaction_XChainClaim": "XChain Claim",
-  "transaction_XChainCommit": "XChain Commit",
-  "transaction_XChainCreateBridge": "XChain Create Bridge",
-  "transaction_XChainCreateClaimID": "XChain Create Claim ID",
-  "transaction_XChainModifyBridge": "XChain Modify Bridge",
-  "transaction_EnableAmendment": "Enable Amendment",
-  "transaction_SetFee": "Set Fee",
-  "transaction_UNLModify": "UNL Modify",
-  "transaction_AccountDelete": "Account Delete",
+  "transaction_type_name_AMMCreate": "AMM Create",
```
Comment (mvadari) [inline public/locales/en-US/translations.json:149]: Why were these changed?

File: src/containers/Accounts/AccountTransactionTable/AccountTransactionTable.tsx [hunk 2]
```diff
@@ -79,7 +79,7 @@ export const AccountTransactionTable = ({
       transactions={transactions}
       loading={loading}
       hasTokensColumn={hasTokensColumn}
-      emptyMessage={emptyMessage && t(emptyMessage)}
+      emptyMessage={emptyMessage && t(emptyMessage as any)}
```
Comment (mvadari) [inline src/containers/Accounts/AccountTransactionTable/AccountTransactionTable.tsx:82]: Should there be `TODO`s on all these casts to come back and fix later?

File: src/i18n/baseConfig.ts [hunk 3]
```diff
@@ -1,9 +1,12 @@
-import i18n, { InitOptions } from 'i18next'
+// @ts-check
```
Comment (mvadari) [inline src/i18n/baseConfig.ts:1]: Why is this necessary in a TS file?

Comment (ckniffen) [reply src/i18n/baseConfig.ts:1]: Good catch.  This was from me running around.

File: src/containers/Accounts/AccountTransactionTable/AccountTransactionTable.tsx (hunk 2 above)
Comment (ckniffen) [reply src/containers/Accounts/AccountTransactionTable/AccountTransactionTable.tsx:82]: I dont think so in this case.

**Review by mvadari — APPROVED** [review]
